pkg/route/validation/validation.go

```diff
@@ -16,11 +17,22 @@ import (
 	kvalidation "k8s.io/apimachinery/pkg/util/validation"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 
+	"github.com/google/go-cmp/cmp"
 	routev1 "github.com/openshift/api/route/v1"
 )
 
 var validateRouteName = apimachineryvalidation.NameIsDNSSubdomain
 
+// Any changes made to this regex shall be reflected in openshift/api for spec.httpHeaders.actions.response[*] of the Route API used for microshift and vice-versa.
```

Comment:
Why not make regex shared between those, in one place?  Where is the `Route API used for microshift` repo?

Reply:
It's not a good idea to have a comment that says any changes made here should also be made in openshift/api.  Just make them in one place, and share it.

Reply:
I don't know why you mention microshift here at all?

Reply:
@JoelSpeed As per https://redhat-internal.slack.com/archives/CCH60A77E/p1671229820312959?thread_ts=1671225380.402839&cid=CCH60A77E  and https://github.com/openshift/enhancements/pull/1296#discussion_r1051221387

Ben Luddy mentioned that "if the api concerns of this enhancement are limited to validation, you should be able to directly change the existing library-go route validation code without introducing any new admission plugins"
and "you might also avoid kubebuilder crd schema generation markers in the o/api types. that's an opportunity to have the CRD-based route api be more strict than OAS".

From overall slack conversation it seems he means that we shall be remove the kube-builder tags validation for this EP from the route in openshift/api as we will be adding validations in the library-go which will be installed in the openshift-apiserver  for OCP and openshift/kubernetes which is used by microshift ?
cc @Miciah @alebedev87 @candita

Reply:
```suggestion
// Any changes made to this regex MUST be duplicated in openshift/api for spec.httpHeaders.actions.response[*] of the Route API for compatibility with MicroShift validation.
```

Reply:
@candita pushed the changes. PTAL

Reply:
// Any changes made to this regex shall be reflected in openshift/api(https://github.com/openshift/api/blob/master/route/v1/types.go & https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go) for Route.spec.httpHeaders.actions[*].response, Route.spec.httpHeaders.actions[*].request, IngressController.spec.httpHeaders.actions[*].response, and IngressController.spec.httpHeaders.actions[*].request of the Route API for the benefit of client-side validation.

Reply:
@candita  is it now clear that this is for client-side validation, not for MicroShift based on the explanation given in the EP ?